### PR TITLE
W-16685129 j17 e2e steps

### PR DESCRIPTION
**Pull request by sathya0**

### CODE REVIEW

File: modules/ROOT/pages/java-support.adoc [hunk 1]
```diff
@@ -33,6 +33,25 @@ The maximum supported version for running your applications is Java 17 for Mule
...
 For more information about Mule release cadences, see xref:release-notes::mule-runtime/lts-edge-release-cadence.adoc[].
 
+== Steps for Upgrade 
+
+Follow these steps to upgrade to Java 17:
+
+. Learn about <<application-performance, application performance>> when upgrading to Java 17. 
+. Conduct an analysis of your current setup to pinpoint compatibility issues or deprecated elements.
+. Create a comprehensive upgrade plan, including testing, rollback strategies, and post-upgrade checks.
+. Schedule downtime and inform stakeholders to minimize business disruption.
```
Comment (marianogonzalez) [inline modules/ROOT/pages/java-support.adoc]: only true for on-prem, and true for any upgrade, not just java 17

File: modules/ROOT/pages/java-support.adoc [hunk 2]
```diff
@@ -33,6 +33,25 @@ The maximum supported version for running your applications is Java 17 for Mule
 
 For more information about Mule release cadences, see xref:release-notes::mule-runtime/lts-edge-release-cadence.adoc[].
 
+== Steps for Upgrade 
+
+Follow these steps to upgrade to Java 17:
+
+. Learn about <<application-performance, application performance>> when upgrading to Java 17. 
+. Conduct an analysis of your current setup to pinpoint compatibility issues or deprecated elements.
```
Comment (marianogonzalez) [inline modules/ROOT/pages/java-support.adoc]: I don't understand the meaning or purpose of this sentense:

Analysis looking for what? 
How is a non java expert supposed to do that?
What is a deprecated element in this context? A Java feature used in custom code? A mule component?

File: modules/ROOT/pages/java-support.adoc [hunk 3]
```diff
@@ -33,6 +33,25 @@ The maximum supported version for running your applications is Java 17 for Mule
 
 For more information about Mule release cadences, see xref:release-notes::mule-runtime/lts-edge-release-cadence.adoc[].
 
+== Steps for Upgrade 
+
+Follow these steps to upgrade to Java 17:
+
+. Learn about <<application-performance, application performance>> when upgrading to Java 17.
```
Comment (marianogonzalez) [inline modules/ROOT/pages/java-support.adoc]: Why is this top of the list?

File: modules/ROOT/pages/java-support.adoc [hunk 4]
```diff
@@ -33,6 +33,25 @@ The maximum supported version for running your applications is Java 17 for Mule
...
 
+== Steps for Upgrade 
+
+Follow these steps to upgrade to Java 17:
+
+. Learn about <<application-performance, application performance>> when upgrading to Java 17. 
+. Conduct an analysis of your current setup to pinpoint compatibility issues or deprecated elements.
+. Create a comprehensive upgrade plan, including testing, rollback strategies, and post-upgrade checks.
+. Schedule downtime and inform stakeholders to minimize business disruption.
+. Implement the upgrade:
```
Comment (marianogonzalez) [inline modules/ROOT/pages/java-support.adoc]: no general upgrade documentation we can point to as a basis?

File: modules/ROOT/pages/java-support.adoc [hunk 5]
```diff
@@ -33,6 +33,25 @@ The maximum supported version for running your applications is Java 17 for Mule
 
 For more information about Mule release cadences, see xref:release-notes::mule-runtime/lts-edge-release-cadence.adoc[].
 
+== Steps for Upgrade 
+
+Follow these steps to upgrade to Java 17:
```
Comment (marianogonzalez) [inline modules/ROOT/pages/java-support.adoc:38]: what about reviewing custom code in all its form?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/java-support.adoc (hunk 3 above)
Comment (sathya0) [reply modules/ROOT/pages/java-support.adoc]: That section has important details regarding Java 17 performance considerations that users should know before doing the upgrade, like prereq info they should know.

File: modules/ROOT/pages/java-support.adoc (hunk 4 above)
Comment (sathya0) [reply modules/ROOT/pages/java-support.adoc]: This page is the general upgrade doc for all things Java 17. In each step, I link to the relevant docs.

File: modules/ROOT/pages/java-support.adoc (hunk 5 above)
Comment (sathya0) [reply modules/ROOT/pages/java-support.adoc:38]: Each of the links has info regarding custom code (custom connectors, custom policies, etc), but I'll add callouts here too

File: modules/ROOT/pages/java-support.adoc (hunk 3 above)
Comment (marianogonzalez) [reply modules/ROOT/pages/java-support.adoc]: why are the performance consideration something the users need to consider before starting? There are a lot of other activities to consider before that

File: modules/ROOT/pages/java-support.adoc (hunk 4 above)
Comment (marianogonzalez) [reply modules/ROOT/pages/java-support.adoc]: can you share a link to preview the rendered result?

File: modules/ROOT/pages/java-support.adoc (hunk 3 above)
Comment (sathya0) [reply modules/ROOT/pages/java-support.adoc]: Can you please list some?

Comment (marianogonzalez) [reply modules/ROOT/pages/java-support.adoc]: Pretty much everything else. Customers can't really evaluate the performance considerations until they actually understand the steps to take and their implications, specially when the upgrade requires changes to their applications.

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/java-support.adoc (hunk 3 above)
Comment (sathya0) [reply modules/ROOT/pages/java-support.adoc]: I'll move the step

**Review by marianogonzalez — APPROVED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Review by bjoson-mulesoft — APPROVED** [review]

**Pull request merged**